substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/c/codegen/CCompilerInvoker.java

```diff
@@ -211,7 +211,7 @@ protected String getDefaultCompiler() {
         @Override
         protected CompilerInfo createCompilerInfo(Scanner scanner) {
             try {
-                while (scanner.findInLine("Apple LLVM version ") == null) {
+                while (scanner.findInLine("Apple.*version ") == null) {
```

Comment:
As there are only two possible values I'd test for both explicitly instead of using a wildcard.